back/engines/free/email_campaigns/app/services/email_campaigns/side_fx_campaign_service.rb

```diff
@@ -8,6 +8,18 @@ def before_create(campaign, user); end
 
     def before_update(campaign, user); end
 
+    def after_update(campaign, _user)
+      # We currently only have one campaign that can be enabled/disabled at phase level,
+      # and we align the respective phase-level values with the campaign enabled when the campaign value is updated.
+      return unless campaign.saved_change_to_enabled? &&
+                    campaign.type == 'EmailCampaigns::Campaigns::ProjectPhaseStarted'
```

Comment:
IMHO `type ==` should be better avoided as it violates the principles of polymorphism.

In this case, it can be replaced with `Phase.update_all` in `after_commit :update_phases, on: :update` in `EmailCampaigns::Campaigns::ProjectPhaseStarted`.

It's good that you've tested it on the controller level, so the tests could stay the same.

Reply:
I see. I was trying to avoid using a model level hook, as it seems unusual in our codebase, and I guessed this was because it can make debugging tricky. i.e. If you manually update the campaign `enabled` and then don't understand why phases get updated.

But I guess a dev can work this out, so I was concerned about an unlikely problem.

Reply:
Both your arguments are very valid, so I leave it up to you to decide which way you like more.

Reply:
I'm now using `instance_of?`, but not really convinced this really addresses the issue you raised.